### ISSUE TITLE
Multiple keypressed registered as single unknown key

### ISSUE THREAD

**Issue by vladdeSV** [vladdeSV/scone#13]
Pressing one (1) button causes an ANSI sequence to be sent. For instance:

* `A` = `97`
* `→` = `27, 91, 67`

The input sequence is read one (1) number at a time, sent from the input thread to the main thread. Eg. → causes the numbers 27, 91, and 67 to be sent to main thread.

The main thread works by: once receiving a number, it waits 1 millisecond, and if another number is sent by then that number also gets appended to the current sequence.

**So if two (2) or more buttons are pressed at the same time, two separate keys can be registered as one (1) key**, because the numbers get sent within the delay.

# (partial) Solution
For every character which get appended, check against the locale to see if that sequence exists. If so, treat it as a continuation of the sequence, otherwise, register it as two separate key presses.

For instance, if `F1 (27, 79, 80)` and `B (98)` are pressed at the same time, either of the following sequences will be produced:

* `27, 79, 80, 98`
* `98, 27, 79, 80`

The input handler should check if the first number is a part of a sequence, and continue until the sequence isn't matched anymore.

For example: Let's assume the sequence `27, 79, 80, 98` is sent.

0. The input handler first checks if `27` matches any sequences.
0. Above is true, so then continue and check if `27, 79` is part of a sequence.
0. Above is true, so then continue and check if `27, 79, 80` is part of a sequence.
0. Above is true, so then continue and check if `27, 79, 80, 98` is part of a sequence.
0. Above is not true, so the handler checks if `27, 79, 80` is a valid sequence.
    * If sequence exists return appropriate key
    * If sequence does not exist, return unknown key
0. Now continue checking numbers, and check if `80` is part of a sequence.
0. Above is true. Last number was found, and return appropriate key or unknown.

---
#### 1. Sequences including sets of sequences
This could in theory break if a button `X` has the sequence `97, 98`, and pressing the `A (97)` and `B (98)` buttons pressed at the same time could register as that button. However, the chance of this effectively happening is (subjective opinion: ) very low.

Pressing `Esc (27)` and `Alt + 8 (91)` and `Shift + A (65)` at the same time could register as the `Up (27, 91, 65)` key.

#### 2. Overlapping sequences
The current proposal would also break if two sequences overlap. An over-simplified example would be: How should the letters `h e l l o m e n` be interpreted; `Hello Men` or `Hell Omen`?

**Comment by vladdeSV** [comment]
*sigh...*

**Comment by vladdeSV** [comment]
I think that a good solution would be to store `InputEvent`s in a tree.

For instance, the following keys and number sequences...
* `tab+shift` = `27,91,90`
* `left+shift` = `27,91,49,59,50,68`

...would be stored in the tree as such:

```
    27
    |
  +-91-+
  |    |
  90   49
  |    | 
shift  59
 tab   |
       50
       |
       68
       |
     shift
     left
```

Where a leaf can either be an `InputEvent` or another number in a sequence. Given a sequence the following things could be happen:
* a sequence is leads to an InputEvent, then that event is returned
* a sequence might be unrecognised or invalid (sequence ends where there is no `InputEvent`), then nothing is returned
* multiple keystrokes might get sent as a sequence. if so, the program will go through all sequences and check against the tree. once it reaches an event, that event is queued, and the program starts over from the beginning.

**Comment by vladdeSV** [comment]
One interesting note I found in my available data is that any sequence of inputs either begin with <kbd>27</kbd> or a number greater than or equal to <kbd>194</kbd>.

**Comment by vladdeSV** [comment]
Fixed by 80edc49854bf99b1255ea21111215a5d1fcf860f
